### PR TITLE
Remove set_inventory_formspec in default, remove delay in sfinv

### LINKED ISSUE #1546: Remove inventory formspec from default/player.lua since sfinv is a part of the game
please remove player:set_inventory_formspec(default.gui_survival_form) from default/player.lua since sfinv is a part of minetest_game.

By the way, please look for a way to proper replace the users inventory formspec. Currently even sfinv must uverride the "default" formspec using minetest.after(0.5..). The on_player_receive_fields is called twice.
Ok, sfinv does have sfinv.enabled that can be set to nil from other mods.

### PR DESCRIPTION
**Pull request by rubenwardy**
Fixes #1546

### CODE REVIEW

File: mods/creative/inventory.lua [hunk 1]
```diff
@@ -33,16 +38,18 @@ function creative.init_creative_inventory(player)
...
 
-	creative.update_creative_inventory(player_name, minetest.registered_items)
+	return player_inventory[player_name]
 end
 
+-- Fills a player's creative inventory with the items listed by tab_content.
+--   calls init_creative_inventory if the inventory does not already exist
+-- @param player_name The name of the player
+-- @param tab_content A key-value list of item names to item defs
 function creative.update_creative_inventory(player_name, tab_content)
```
Comment (rubenwardy) [inline mods/creative/inventory.lua]: would be good to change this to just `player` - however could require backwards compat code (if it's used)

File: mods/creative/inventory.lua [hunk 2]
```diff
@@ -1,5 +1,10 @@
 local player_inventory = {}
 
+-- Creates a blank detached inventory to be filled with contents later
```
Comment (kaeza) [inline mods/creative/inventory.lua]: :+1: for doc-comments, but is this LuaDoc/LDoc, Doxygen, or what?

For LuaDoc/LDoc it should be three dashes (`---`). Not sure about Doxygen.

Comment (rubenwardy) [inline mods/creative/inventory.lua]: I need to investigate this. I personally think it's worth autogenerating game_api.txt from Luadoc/Doxygen

Comment (sofar) [inline mods/creative/inventory.lua]: I wouldn't mind `ldoc`, but please, no doxygen :)

**Comment by sofar** [comment]
This looks good, but, rebase needed.

**Comment by paramat** [comment]
@rubenwardy should be in release? Maybe you can rebase and merge?

**Comment by paramat** [comment]
rubenwardy states this is not essential for release, he also has no time for MT until Friday, maybe someone can rebase, approve and merge this?

**Comment by paramat** [comment]
Should fix #1692

**Comment by paramat** [comment]
Replaced by #1738

**Comment by paramat** [comment]
#1738 merged.

**Pull request closed without merging**